src/main/java/com/openshift/jenkins/plugins/pipeline/dsl/OpenShiftDeploymentVerifier.java

```diff
@@ -87,57 +97,90 @@ public BuildStepMonitor getRequiredMonitorService() {
-        }
-
-        @Override
-        public String getFunctionName() {
-            return "openshiftVerifyDeployment";
-        }
-
-        @Override
-        public String getDisplayName() {
-            return DISPLAY_NAME;
-        }
-
-        @Override
-        public Step newInstance(Map<String, Object> arguments) throws Exception {
-            if (!arguments.containsKey("deploymentConfig") && !arguments.containsKey("depCfg"))
-            	throw new IllegalArgumentException("need to specify deploymentConfig");
-            Object depCfg = arguments.get("deploymentConfig");
-            if (depCfg == null || depCfg.toString().length() == 0)
-            	depCfg = arguments.get("depCfg");
-            if (depCfg == null || depCfg.toString().length() == 0)
-            	throw new IllegalArgumentException("need to specify deploymentConfig");
-            OpenShiftDeploymentVerifier step = new OpenShiftDeploymentVerifier(depCfg.toString());
-            
-            if (arguments.containsKey("waitTime")) {
-            	Object waitTime = arguments.get("waitTime");
-            	if (waitTime != null)
-            		step.setWaitTime(waitTime.toString());
-            }
-            if (arguments.containsKey("replicaCount")) {
-            	Object replicaCount = arguments.get("replicaCount");
-            	if (replicaCount != null)
-            		step.setReplicaCount(replicaCount.toString());
-            }
-            if (arguments.containsKey("verifyReplicaCount")) {
-            	Object verifyReplicaCount = arguments.get("verifyReplicaCount");
-            	if (verifyReplicaCount != null)
-            		step.setVerifyReplicaCount(verifyReplicaCount.toString());
-            }
-            
-            ParamVerify.updateDSLBaseStep(arguments, step);
-            return step;
-        }
-    }
-
-
+	public static class DescriptorImpl extends AbstractStepDescriptorImpl implements IOpenShiftTimedStepDescriptor {
+		private String wait;
+		private String waitUnit = "sec";
+
+		public DescriptorImpl() {
+			super(OpenShiftDeploymentVerifierExecution.class);
+		}
+
+		@Override
+		public String getFunctionName() {
+			return "openshiftVerifyDeployment";
+		}
+
+		@Override
+		public String getDisplayName() {
+			return DISPLAY_NAME;
+		}
+
+		@Override
+		public Step newInstance(Map<String, Object> arguments) throws Exception {
+			if (!arguments.containsKey("deploymentConfig") && !arguments.containsKey("depCfg"))
+				throw new IllegalArgumentException("need to specify deploymentConfig");
+			Object depCfg = arguments.get("deploymentConfig");
+			if (depCfg == null || depCfg.toString().length() == 0)
+				depCfg = arguments.get("depCfg");
+			if (depCfg == null || depCfg.toString().length() == 0)
+				throw new IllegalArgumentException("need to specify deploymentConfig");
+			OpenShiftDeploymentVerifier step = new OpenShiftDeploymentVerifier(depCfg.toString());
+
+			if (arguments.containsKey("waitTime")) {
```

Comment:
leverage the new default method on IOpenShiftTimedStepDescriptor called for earlier